src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java

```diff
@@ -171,50 +172,84 @@ private void setUpdateFilingApiData(FilingApi filing, String transactionId, Stri
                     .firstName(data.getFirstName())
                     .middleNames(data.getMiddleNames())
                     .lastName(data.getLastName())
                     .formerNames(data.getFormerNames());
         }
-        if (data.getNationalityHasBeenUpdated() == null || data.getNationalityHasBeenUpdated()) {
+        if (nationalityHasBeenUpdated) {
             dataBuilder = dataBuilder.nationality1(data.getNationality1())
                     .nationality2(data.getNationality2())
                     .nationality3(data.getNationality3());
         }
-        if (data.getOccupationHasBeenUpdated() == null || data.getOccupationHasBeenUpdated()) {
+        if (occupationHasBeenUpdated) {
             dataBuilder = dataBuilder.occupation(data.getOccupation());
         }
-        if (data.getCorrespondenceAddressHasBeenUpdated() == null || data.getCorrespondenceAddressHasBeenUpdated()) {
+        if (correspondenceAddressHasBeenUpdated) {
             dataBuilder = dataBuilder.isServiceAddressSameAsRegisteredOfficeAddress(data.getIsServiceAddressSameAsRegisteredOfficeAddress());
             if (!Boolean.TRUE.equals(data.getIsServiceAddressSameAsRegisteredOfficeAddress())) {
                 dataBuilder = dataBuilder.serviceAddress(data.getServiceAddress());
             }
         }
-        if (data.getResidentialAddressHasBeenUpdated() == null || data.getResidentialAddressHasBeenUpdated()) {
+        if (residentialAddressHasBeenUpdated) {
             dataBuilder = dataBuilder.isHomeAddressSameAsServiceAddress(data.getIsHomeAddressSameAsServiceAddress());
             if (!Boolean.TRUE.equals(data.getIsHomeAddressSameAsServiceAddress())) {
                 dataBuilder = dataBuilder.residentialAddress(data.getResidentialAddress());
             }
         }
+        dataBuilder = dataBuilder.countryOfResidence(getCountryOfResidence(data, appointment, residentialAddressHasBeenUpdated, correspondenceAddressHasBeenUpdated));
 
-        final var enhancedOfficerFiling = OfficerFiling.builder(officerFiling)
-                .createdAt(officerFiling.getCreatedAt())
-                .updatedAt(officerFiling.getUpdatedAt())
-                .data(dataBuilder.build())
-                .build();
-        var filingData = filingAPIMapper.map(enhancedOfficerFiling);
-        var dataMap = MapHelper.convertObject(filingData, PropertyNamingStrategies.SNAKE_CASE);
-        logger.debugContext(transactionId, "Created update filing data for submission", new LogHelper.Builder(transaction)
-                .withFilingId(filingId)
-                .build());
+        return dataBuilder;
+    }
 
-        filing.setData(dataMap);
-        setCh01DescriptionFields(filing, enhancedOfficerFiling.getData(), appointment);
+    private String getCountryOfResidence(OfficerFilingData data, AppointmentFullRecordAPI appointment, boolean residentialAddressHasBeenUpdated, boolean correspondenceAddressHasBeenUpdated) {
```

Comment:
getChangedCoutryOfResidence? to make it clearer that we are interested in any change of address that changes the country of residence.